### PR TITLE
Fix non-matching items and post type matching

### LINKED ISSUE #5575: Nav jumps to Tools/Utilities after creating new product or order
**Describe the bug**
After you create a new product or order, the new nav jumps to **Tools > Utilities**.

**To Reproduce**
1. Create a new product or order.
2. Click "Publish".
3. See nav jump to **Tools > Utilities**

**Expected behavior**
The navigation should stay on the existing item, `Add product` or `Add order`.

**Screenshots**
Screen Shot 2020-11-09 at 3 45 25 PM

Screen Shot 2020-11-09 at 3 32 39 PM

**Desktop (please complete the following information):**

* Tested on Chrome, Firefox, and Vivaldi.

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #5575 

* Avoids matching the last item if no matches exist.
* Matches post type edit pages to the respective post type "all" menu item.

### Screenshots
Screen Shot 2020-11-10 at 3 52 11 PM

### Detailed test instructions:

1. Enable the navigation feature.
1. Create or edit an existing order.
1. Note the active menu item is correct.
1. Check that no regressions have occurred with other menu item's active status.

### CODE REVIEW

File: src/Features/Navigation/Menu.php [hunk 1]
```diff
@@ -538,6 +538,7 @@ public function enqueue_data( $menu ) {
 
 		$data = array(
 			'menuItems' => self::get_prepared_menu_item_data(),
+			'postType'  => isset( $_GET['post'] ) ? get_post_type( $_GET['post'] ) : null,
 		);
 
 		$paul = wp_add_inline_script( WC_ADMIN_APP, 'window.wcNavigation = ' . wp_json_encode( $data ), 'before' );
```
Comment (joelclimbsthings) [inline src/Features/Navigation/Menu.php:544]: I know this isn't part of the PR, but I couldn't _not_ ask about `$paul` here.

Comment (joshuatf) [reply src/Features/Navigation/Menu.php:544]: 😆 I hadn't even noticed this.  @psealock must have slipped in this easter egg when we transitioned from the nav repo.

**Review by joelclimbsthings — APPROVED** [review]
Tested & works for me :+1:

File: src/Features/Navigation/Menu.php (hunk 1 above)
Comment (psealock) [reply src/Features/Navigation/Menu.php:544]: 😆 Wait, you don't name variables after yourself!? I wrote that specifically to avoid it getting committed 😨

**Review by psealock — APPROVED** [review]
code changes LGTM :shipit:

**Comment by timmyc** [comment]
cherry picked to `release/1.7.0` in https://github.com/woocommerce/woocommerce-admin/commit/c6b531cfa63acc63acb048cc12a5c081b783b2cc

**Pull request merged**